Added video classification TFLite code.

khanhlvg opened the pull request:
This is the first pull request to add a video classification sample to the TFLite Reference app repo.

This pull request contains:
* VideoClassifier: The wrapper class to run video classification with an interface that resembles the TFLite Task Library.
* VideoClassifierTest: The test code of the video classifier.
* CalculateUtils: Some shared utils.

khanhlvg commented:
@mbrenon PTAL as well. Thanks!

lu-wang-g commented on lite/examples/video_classification/android/app/src/main/java/org/tensorflow/lite/examples/videoclassification/ml/VideoClassifier.kt:
```diff
@@ -0,0 +1,255 @@
+            }
+
+            fun setMaxResult(maxResults: Int): Builder {
+                if (maxResults == 0) {
```
`maxResults == 0` => `maxResults <= 0` make sure the value is greater than 0?

khanhlvg replied:
Fixed. I changed to allow maxResults = -1 to indicate that the users want all result.

lu-wang-g commented on lite/examples/video_classification/android/app/src/main/java/org/tensorflow/lite/examples/videoclassification/ml/VideoClassifier.kt:
```diff
@@ -0,0 +1,255 @@
+    }
+
+    /**
+     * Convert input bitmap to TensorImage and normalize.
```
Update this java doc to fit postprocessOutputLogits?

khanhlvg replied:
Fixed.

lu-wang-g commented on lite/examples/video_classification/android/app/src/main/java/org/tensorflow/lite/examples/videoclassification/ml/VideoClassifier.kt:
```diff
@@ -0,0 +1,255 @@
+     */
+    private fun postprocessOutputLogits(logitsByteBuffer: ByteBuffer): MutableList<Category> {
+        // Convert ByteBuffer to FloatArray.
+        val tensorBuffer = TensorBuffer.createFixedSize(
```
It seems to be overkilled to use TensorBuffer to convert the ByteBuffer into a float array. Does it work to project the ByteBuffer to FloatBuffer, and then get the float array? Alternatively, you can create a TensorBuffer to store the output, meaning passing the TensorBuffer to Interpreter instead of using ByteBuffer.

khanhlvg replied:
Fixed by converting ByteBuffer directly to FloatArray. As the model outputs contain INT32 tensor, I can't use TensorBuffer for all outputs because INT32 isn't supported yet.

khanhlvg commented:
@lu-wang-g Thanks for reviewing! I've fixed the feedback and made some minor changes:
* Changed the model used from float model to int8 quantized model to reduce inference time.
* Changed the test video as the previous one didn't work well with int8 model.

lu-wang-g reviewed (approved):
Thanks for updating the code, Khanh!

khanhlvg commented:
Submitted in https://github.com/tensorflow/examples/commit/86e95a594ffd7a74016a2b15991d2294725d7e28
